File: programming 1/lab 6/src/main/java/common/utils/SerializationUtils.java
package common.utils;

import java.io.*;

public class SerializationUtils {
    public static byte[] serialize(Object obj) {
        try (ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
             ObjectOutputStream objectStream = new ObjectOutputStream(byteStream)) {
            objectStream.writeObject(obj);
            objectStream.flush();
            return byteStream.toByteArray();
        } catch (IOException e) {
            throw new RuntimeException("Произошла ошибка при сериализации данных: " + e);
        }
    }

    public static Object deserialize(byte[] data) {
        try (ByteArrayInputStream byteStream = new ByteArrayInputStream(data);
             ObjectInputStream objectStream = new ObjectInputStream(byteStream)) {
            return objectStream.readObject();
        } catch (IOException e) {
            throw new RuntimeException("Произошла ошибка при десериализации данных: " + e);
        } catch (ClassNotFoundException e) {
            throw new RuntimeException("Класс для десериализации данных не найден: " + e);
        }
    }
}

File: programming 1/lab 7/src/main/java/common/utils/SerializationUtils.kt
package common.utils

import java.io.*

object SerializationUtils {
    fun  serialize(obj: Any?): ByteArray {
        try {
            ByteArrayOutputStream().use { byteStream ->
                ObjectOutputStream(byteStream).use { objectStream ->
                    objectStream.writeObject(obj)
                    objectStream.flush()
                    return byteStream.toByteArray()
                }
            }
        } catch (e: IOException) {
            throw RuntimeException("Произошла ошибка при сериализации данных: $e")
        }
    }

    fun deserialize(data: ByteArray?): Any {
        try {
            ByteArrayInputStream(data).use { byteStream ->
                ObjectInputStream(byteStream).use { objectStream ->
                    return objectStream.readObject()
                }
            }
        } catch (e: IOException) {
            throw RuntimeException("Произошла ошибка при десериализации данных: $e")
        } catch (e: ClassNotFoundException) {
            throw RuntimeException("Класс для десериализации данных не найден: $e")
        }
    }
}
